src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -23,7 +23,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
                         ' It is not permissible to access the Left property of a BoundUserDefinedBinaryOperator unconditionally,
                         ' because that property can throw an exception if the operator expression is semantically invalid.
                         ' Fetching the left operand through IBinaryOperatorExpression is safe.
-                        If rightOperatorBinary.LeftOperand Is value.LeftOnTheRightOpt Then
+                        If rightOperatorBinary.LeftOperand Is Create(value.LeftOnTheRightOpt) Then
```

Comment:
The code should be based of off bound nodes, content of bound nodes determines the result we are calculating. We should not be taking dependency on today's implementation of IOperation, which might or might not change in the future, especially around placeholder nodes (this is what LeftOnTheRightOpt is). As for the code reuse, we can add a helper to the BoundUserDefinedBinaryOperator and reuse it in two places.